cosmoz-data-nav.js

```diff
@@ -488,52 +483,38 @@
 			};
 		},
 
-		_toggleInstance(element, index, incomplete) {
-			if (element.__instanceActive === !incomplete) {
-				return;
-			}
-
-			const baseProps = this._getBaseProps(index);
+		_getElement(index) {
+			return this._elements[index % this.elementsBuffer];
+		},
 
-			element.__instanceActive = !incomplete;
+		_resetElement(index) {
+			const element = this._getElement(index);
 
-			if (element.__incomplete) {
-				element.__incomplete._showHideChildren(!incomplete);
-				Object.assign(element.__incomplete, baseProps);
+			if (!element || element._reset) {
+				return;
 			}
 
-			if (element.__instance) {
-				element.__instance._showHideChildren(incomplete);
-				Object.assign(element.__instance, baseProps);
-			}
-		},
+			const item = this.items[index];
 
-		/**
-		 * Forwards an item from `items` property to a template instance.
-		 *
-		 * @param  {HTMLElement} element The element
-		 * @param  {Object}      item     The item to forward
-		 * @return {void}
-		 */
-		_forwardItem(element, item) {
-			if (element.item === item) {
-				// already rendered
+			if (!this.isIncompleteFn(item) && element.item === item) {
 				return;
 			}
+			element._reset = true;
+			console.log('resetting item', index);
 
-			this._removeInstance(element.__instance);
+			const baseProps = this._getBaseProps(index),
+				incomplete = element.__incomplete,
+				instance = element.__instance;
```

Comment:
Since resetElement only will be run once per element, we shouldn't do any work on instance, only incomplete.

Reply:
Element can be complete but have a different old item and needs to be in reset state until it will be templatized again.

Reply:
Yes. I am not filtering in synchronize because _resetElement will check that.